shark_turbine/kernel/wave/register_analysis.py

```diff
@@ -51,5 +87,10 @@ def determine_register_shape(trace: CapturedTrace | fx.Graph) -> None:
     register_nodes = trace.walk(lambda node: isinstance(get_custom(node), NewRegister))
     if not register_nodes:
         return
+    vector_map = {}
+    for c in constraints:
```

Comment:
This seems like a useful piece of code. Do you think you can put it as a function in utils?

Reply:
SG! Done!